snuba/consumers/consumer_builder.py

```diff
@@ -93,12 +96,20 @@ def __init__(
             .topic
         )
 
+        if slice_id is not None:
+            # there should be some sort of assertion to check
+            # whether the slice_id passed is valid for this storage
+            physical_topic = SLICED_KAFKA_TOPIC_MAP[(topic.value, slice_id)]
```

Comment:
Can you put this behind a function? this makes an assumption that the way logical topics and physical topics are mapped and is harder to test in case it changes

Reply:
in fact you can save a lot of if statements in this function if said function can take a `None` slice_id and just return the `topic.value`